pacfish_matlab/@kwave_adapter/kwave_adapter.m

```diff
@@ -325,37 +385,57 @@
+                            % point of the line in the x and y
+                            % directions, squaring each and summing
+                            % them and lastly square rooting the sum
+                            % which is using pythagoras to determine
+                            % line length assuming there is a change in
+                            % both x and y
+                            % If change is in 1 dimension only the other distance
+                            % would be 0 and therefore the distance of the
+                            % non-0 direction would not be considered
+                            % This length is stored as elem.width as that's
+                            % how rectangle elements would store their length
+                            elem.width = sqrt((dist(elem.start_point(1),elem.end_point(1)))^2+(dist(elem.start_point(2),elem.end_point(2)))^2);
+                            
+                            % The project uses sohcahtoa
+                            % to determine the angle used for
+                            % orientation 
+                            % Specifically uses cos theta =
+                            % adjacent/hypotenuse
+                            % adjacent = y direction distance
+                            % hypotenuse = element length (elem.width)
+                            costheta = dist(elem.start_point(2),elem.end_point(2))/elem.width;
+                            % Sets the orientation to 180 - the cos inverse
+                            % of cos theta to fix any directional issues
+                            % the normal inverse would cause
+                            elem.orientation = acosd(costheta);
+                            
+                            % Width is the line height determined
+                            % earlier
+                            % Stored as elem.length as that's where
+                            % rectangle elements store their width
+                            elem.length = obj.lineheight;
                             % Orientation doesn't exist for lines
                              % Geometry only requires 2 dimensions here x
                             % and y as input is 2D
-                            device_struct.detectors.(index).detector_geometry = [elem.length elem.width];
+                            
+                            device_struct.detectors.(index).detector_orientation = elem.orientation; % Sets element orientation
+                            
+                            device_struct.detectors.(index).detector_geometry = [elem.length elem.width]; 
+
+                        end
                 end
                 
                 
                 
-                %% FIXME: we need a FOV definition.
-                if any(obj.fov)
-                    device_struct.general.field_of_view = obj.fov;
+                %% FIXME: we need a FOV definition. - FIXED

```

Comment:
we can delete this line :D

Reply:
Not sure how to reply to the double check all the maths comment but yes please if possible

Reply:
> we can delete this line :D



> Looking good! Do you want me to double-check all the maths?

Oh nvm that's how anyway yes please if you can double check the maths just in case I made errors